### PR TITLE
Implement highlevel svelte components

### PR DESCRIPTION
**Pull request by scholarsmate**

### CODE REVIEW

File: src/dataEdit/dataEditWebView.ts [hunk 1]
```diff
@@ -434,9 +434,9 @@ export class DataEditWebView implements vscode.Disposable {
     this.currentViewportId = viewportId
     await viewportSubscribe(this.panel, this.currentViewportId)
 
-    const vpResponse = await getViewportData(this.currentViewportId)
-    const data = vpResponse.getData_asU8()
-    const display = Buffer.from(data).toString('hex')
+    // const vpResponse = await getViewportData(this.currentViewportId)
+    // const data = vpResponse.getData_asU8()
+    // const display = Buffer.from(data).toString('hex')
```
Comment (scholarsmate) [inline src/dataEdit/dataEditWebView.ts:439]: Don't comment out code, just remove it.

**Comment by scholarsmate** [comment]
Add license blurbs to the new files, fix the formatting with prettier, and create a single-commit patch to pass CI.

**Comment by scholarsmate** [comment]
Somewhere in the refactoring, `display.ts` is failing to compile with:

```
src/svelte/src/utilities/display.ts:39:25 - error TS2304: Cannot find name 'HTMLTextAreaElement'.

39 ): ViewportReferences | HTMLTextAreaElement {
                           ~~~~~~~~~~~~~~~~~~~

src/svelte/src/utilities/display.ts:41:8 - error TS2584: Cannot find name 'document'. Do you need to change your target library? Try changing the 'lib' compiler option to include 'dom'.

41     ? (document.getElementById(viewport) as HTMLTextAreaElement)
```

I've moved the refactoring into a different branch until I can figure it out.

File: src/svelte/src/components/dataEditor.svelte [hunk 2]
```diff
@@ -248,7 +248,10 @@ limitations under the License.
         break
 
       case MessageCommand.editorOnChange:
-        $editorSelection = msg.data.display
+        if($editMode === EditByteModes.Multiple)
+          $editorSelection = msg.data.display
+
+        console.log("editorOnChange recieved: ", msg.data.display)
```
Comment (scholarsmate) [inline src/svelte/src/components/dataEditor.svelte:254]: Remove this.

File: src/svelte/src/stores/index.ts [hunk 3]
```diff
@@ -80,6 +80,13 @@ export const editByte = derived(
     $viewportData,
     $selectionData,
   ]) => {
+    console.log("EditByte: ",
+      $displayRadix,
+      $focusedViewportId,
+      $editMode,
+      $viewportData,
+      $selectionData
+    )
```
Comment (scholarsmate) [inline src/svelte/src/stores/index.ts:89]: Remove this.

File: src/svelte/src/components/DataDisplays/DataViewports.svelte [hunk 4]
```diff
@@ -214,6 +201,9 @@ limitations under the License.
 <textarea
   class={$UIThemeCSSClass + ' physical'}
   class:locked={$selectionData.active}
+  class:hexWidth={$displayRadix === RadixOptions.Hexidecimal}
+  class:decoctWidth={$displayRadix === RadixOptions.Decimal || $displayRadix === RadixOptions.Octal}
+  class:binWidth={$displayRadix === RadixOptions.Binary}
```
Comment (scholarsmate) [inline src/svelte/src/components/DataDisplays/DataViewports.svelte:206]: Seems like we can just apply one of these classes and not have them all here with true/false values.  We can re-use that code on lines 218 - 220.

Comment (stricklandrbls) [reply src/svelte/src/components/DataDisplays/DataViewports.svelte]: I'll look this up. From what I remember about the Svelte tutorials, the `class:` directive only receives _truthy_ input towards which classes to obtain. Let me double check though.

**Comment by stricklandrbls** [comment]
> Somewhere in the refactoring, `display.ts` is failing to compile with:
> 
> 
> 
> ```
> 
> src/svelte/src/utilities/display.ts:39:25 - error TS2304: Cannot find name 'HTMLTextAreaElement'.
> 
> 
> 
> 39 ): ViewportReferences | HTMLTextAreaElement {
> 
>                            ~~~~~~~~~~~~~~~~~~~
> 
> 
> 
> src/svelte/src/utilities/display.ts:41:8 - error TS2584: Cannot find name 'document'. Do you need to change your target library? Try changing the 'lib' compiler option to include 'dom'.
> 
> 
> 
> 41     ? (document.getElementById(viewport) as HTMLTextAreaElement)
> 
> ```
> 
> 
> 
> I've moved the refactoring into a different branch until I can figure it out.

Sometimes I get weird errors like that and the only thing that resolved it is to close VSCode and open it again. I've yet to figure it out. It happens a lot with the 'tsconfig' file.

**Comment by scholarsmate** [comment]
> > Somewhere in the refactoring, `display.ts` is failing to compile with:
> > ```
> > 
> > src/svelte/src/utilities/display.ts:39:25 - error TS2304: Cannot find name 'HTMLTextAreaElement'.
> > 
> > 
> > 
> > 39 ): ViewportReferences | HTMLTextAreaElement {
> > 
> >                            ~~~~~~~~~~~~~~~~~~~
> > 
> > 
> > 
> > src/svelte/src/utilities/display.ts:41:8 - error TS2584: Cannot find name 'document'. Do you need to change your target library? Try changing the 'lib' compiler option to include 'dom'.
> > 
> > 
> > 
> > 41     ? (document.getElementById(viewport) as HTMLTextAreaElement)
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > I've moved the refactoring into a different branch until I can figure it out.
> 
> Sometimes I get weird errors like that and the only thing that resolved it is to close VSCode and open it again. I've yet to figure it out. It happens a lot with the 'tsconfig' file.

@stricklandrbls, so the source of the problem was in de-duplicating the call to `export function radixBytePad(radix: number): number` which appears in 2 places, `utilities/display.ts` and in `dataEditor/dataEditorClient.ts`.  The problem is that the top-level buils system is responsible for `dataEditor/dataEditorClient.ts` whereas the `svelte` stuff has it's own build system, with a different `tsconfig.json` that includes the `DOM` lib needed for `HTMLTextAreaElement` and friends.  So long story short, we can't currently import certain code across that boundary because they are being built differently.  I hope that clears it up.

**Comment by stricklandrbls** [comment]
@scholarsmate 
I'm seeing an issue with starting up a data editor instance that evolved into a major issue.

**Originally**, I could open a selected file into the data editor but the server heartbeat information display footer was showing: `Powered by ΩEdit vunknown port 0` and all of the statistics were displaying `0` or `0.0`.

---

**Currently**, whenever I start up a data editor instance all I'm seeing are two VSCode error notifications:

image

No log file is even being created in `$HOME/.local/share/omega_edit/`.

**Pull request closed without merging**